### PR TITLE
Harvester: Implement Alertmanager

### PR DESCRIPTION
**Pull request by DaiYuzeng**
### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/2643

### CODE REVIEW

File: pkg/harvester/l10n/en-us.yaml [hunk 1]
```diff
@@ -769,4 +781,9 @@ typeLabel:
     {count, plural,
       one { Cloud Config Template }
       other { Cloud Config Templates }
+    }
+  harvesterhci.io.monitoring.alertmanagerconfig: |-
+    {count, plural,
+      one { AlertmanagerConfig }
```
Comment (mantis-toboggan-md) [inline pkg/harvester/l10n/en-us.yaml:787]: Same as above, `Alertmanager Config` and `Alertmanager Configs`

File: pkg/harvester/l10n/en-us.yaml [hunk 2]
```diff
@@ -684,6 +686,16 @@ harvester:
       label: Protocols
     ciphers:
       label: Ciphers
+  
+  monitoring:
+    configuration:
+      label: Configuration
+    alertmanagerConfig:
+      label: AlertmanagerConfigs
```
Comment (mantis-toboggan-md) [inline pkg/harvester/l10n/en-us.yaml:694]: This should probably be `Alertmanager Configs`

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]
I think it would be better to disable the 'create' button when the banner saying configs can't be created is shown. As it is now, it's a little confusing: 
Screen Shot 2022-09-13 at 2 57 12 PM

Aside from this and the translation comments, the PR looks/works good.

**Review by WuJun2016 — CHANGES_REQUESTED** [review]
image

Delete `Cancel` button, other LGTM

**Comment by WuJun2016** [comment]
other LGTM，

The Disable `Create` button is not [working](https://github.com/harvester/dashboard/pull/411/files#diff-8b795e0d77fe25ef54f4c626b129005448c171d718f61f6dc920ee1a8290eedbR42) properly,   I think it's caused here https://github.com/rancher/dashboard/blob/c9a993f197973d9568af674742e338aa11cd996f/shell/store/type-map.js#L1547
 
I think we should override the value of `state.typeOptions[idx]` use `obj`,  

`Object.assign(obj, state.typeOptions[idx])  => Object.assign(state.typeOptions[idx], obj)`

@mantis-toboggan-md    Can you help confirm it ?  thanks

**Comment by WuJun2016** [comment]
> other LGTM，
> 
> The Disable `Create` button is not [working](https://github.com/harvester/dashboard/pull/411/files#diff-8b795e0d77fe25ef54f4c626b129005448c171d718f61f6dc920ee1a8290eedbR42) properly, I think it's caused here https://github.com/rancher/dashboard/blob/c9a993f197973d9568af674742e338aa11cd996f/shell/store/type-map.js#L1547
> 
> I think we should override the value of `state.typeOptions[idx]` use `obj`,
> 
> `Object.assign(obj, state.typeOptions[idx]) => Object.assign(state.typeOptions[idx], obj)`
> 
> @mantis-toboggan-md Can you help confirm it ? thanks

Test method: 
switch between disabled and enabled, the Create button on the alertmanagerconfig page should be correctly hidden or displayed

image

**Pull request merged**